nanoFramework.Hardware.Esp32.Rmt/ReceiverChannelSettings.cs

```diff
@@ -0,0 +1,109 @@
+﻿//
+// Copyright (c) .NET Foundation and Contributors
+// See LICENSE file in the project root for full license information.
+//
+
+using System;
+
+namespace nanoFramework.Hardware.Esp32.Rmt
+{
+    /// <summary>
+    /// <see cref="ReceiverChannel"/>'s settings class.
+    /// </summary>
+    /// <remarks>
+    /// All changes made to properties of this class are ignored after the <see cref="ReceiverChannel"/> is initialized.
+    /// The equivalent properties in the channel instance can be used to make on-the-fly changes to the configurations.
+    /// </remarks>
+    public sealed class ReceiverChannelSettings : RmtChannelSettings
+    {
+        private ushort _idleThreshold;
+        private bool _enableFilter;
+        private byte _filterThreshold;
+        private TimeSpan _receiveTimeout;
+
+        /// <summary>
+        /// Gets or sets the idle threshold after which the receiver will go into idle mode 
+        /// and <see cref="RmtCommand"/>s are copied into the ring buffer and availble to your code.
+        /// </summary>
+        /// <remarks>
```

Comment:
you can add as well the argument exception in the description. That's nice for developers as we are removing the text from the code.